pkg/provider/aws/execution_unit.go

```diff
@@ -0,0 +1,73 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/klothoplatform/klotho/pkg/config"
+	"github.com/klothoplatform/klotho/pkg/core"
+	"github.com/klothoplatform/klotho/pkg/provider/aws/resources"
+	"github.com/klothoplatform/klotho/pkg/provider/aws/resources/ecr"
+	"github.com/klothoplatform/klotho/pkg/provider/aws/resources/iam"
+	"github.com/klothoplatform/klotho/pkg/provider/aws/resources/lambda"
+	"go.uber.org/zap"
+)
+
+func (a *AWS) GenerateExecUnitResources(unit *core.ExecutionUnit, dag *core.ResourceGraph) error {
+	log := zap.S()
+
+	execUnitCfg := a.Config.GetExecutionUnit(unit.ID)
+
+	// See if we have already created an ecr repository for the app and if not create one, otherwise add a ref to this exec unit
+	repo := dag.GetResource(ecr.GenerateRepoId(a.Config.AppName))
+	if repo == nil {
+		repo = ecr.NewEcrRepository(a.Config.AppName, unit.Provenance())
+		dag.AddResource(repo)
+	} else {
+		repo, ok := repo.(*ecr.EcrRepository)
+		if !ok {
+			return fmt.Errorf("expected resource with id, %s, to be ecr repository", repo.Id())
+		}
+		repo.ConstructsRef = append(repo.ConstructsRef, unit.Provenance())
+	}
+
+	// Create image and make it dependent on the repository
+	image := ecr.NewEcrImage(unit, a.Config.AppName)
+	dag.AddResource(image)
+	dag.AddDependency(repo, image)
+
+	// Create and add role
+	role := iam.NewIamRole(a.Config.AppName, fmt.Sprintf("%s-ExecutionRole", unit.ID), unit.Provenance(), GetAssumeRolePolicyForType(execUnitCfg))
+
+	dag.AddResource(role)
+
+	switch execUnitCfg.Type {
+	case Lambda:
+		lambda := lambda.NewLambdaFunction(unit, a.Config.AppName, role)
+		dag.AddResource(lambda)
+		dag.AddDependency(role, lambda)
+		dag.AddDependency(image, lambda)
+		return nil
+	case Eks:
+	case Ecs:
```

Comment:
should we comment these two out for now, so that they'll result in the Error message? (go cases don't fall through)

Reply:
Yeah i need to likely go clean this all up, some of this and the other switch statements were just brain dumped that i left over